docs/contracts/staking-router.md

```diff
@@ -3,9 +3,843 @@
+StakingRouter is a registry of staking modules, each encapsulating a certain validator subset, e.g. the Lido DAO-curated staking module. The contract allocates stake to modules, distributes protocol fees, and tracks relevant information.
 
-See also:
+## What is StakingRouter?
 
-- [LIP-20. Staking Router](https://github.com/lidofinance/lido-improvement-proposals/blob/develop/LIPS/lip-20.md)
-- [Staking Router Explained](https://twitter.com/LidoFinance/status/1624071971011977219?s=20) (twitter thread)
+StakingRouter is a top-level controller contract for staking modules. Each staking module is a contract that, in turn, manages its own subset of validators, e.g. the [Curated](https://etherscan.io/address/0x55032650b14df07b85bF18A3a3eC8E0Af2e028d5) staking module is a set of Lido DAO-vetted node operators. Such modular design opens the opportunity for anyone to build a staking module and join the Lido staking platform, including permissionless community stakers, DVT-enabled validators or any other validator subset, technology or mechanics.
+
+StakingRouter performs a number of functions, including:
+- a registry of staking modules,
+- allocating stake to modules, and
+- distributing protocol fees.
+
+## Module Management
+
+### Registering a module
+
+Modules are registered with StakingRouter through the Lido DAO voting process. To be considered by the governance, the applying module contract should implement the appropriate module interface, meet security requirements, and have a fee structure aligned with the Lido protocol sustainability. Once voted in, the module starts receiving stake and protocol fees.
+
+Staking modules are registered using the `addStakingModule` function, providing details such as:
+- the module's name: a human-readable name;
+- address of the deployed staking module contract;
+- target share, a relative hard cap on deposits within Lido;
+- module fee, a percentage of staking rewards to be awarded to the module,
+- and treasury fee, a percentage of staking rewards to be directed to the protocol treasury.
+
+### Pausing modules
+
+Each staking module has a status: a state that determines whether the module can perform deposits and receive rewards:
+- `Active`, can make deposits and receives rewards,
+- `DepositsPaused`, deposits are not allowed but receives rewards,
+- `Stopped`, cannot make deposits and does not receive rewards.
+
+```solidity
+enum StakingModuleStatus {
+	Active, // deposits and rewards allowed
+	DepositsPaused, // deposits NOT allowed, rewards allowed
+	Stopped // deposits and rewards NOT allowed
+}
+```
+
+### Exited and stuck validators
+
+When the withdrawal requests exceed the Lido reserved buffer, the protocol signals to node operators to start exiting validators to cover the withdrawals. In this connection, StakingRouter distinguishes two types of validator states:
```

Comment:
maybe
'When the withdrawal requests demand exceed buffered ether sitting in Lido together with projected rewards, ..."